### PR TITLE
ProtoMsg moves in.

### PR DESCRIPTION
**Pull request by merlin-northern**
ProtoMsg moves in.

goes with:

* https://github.com/mendersoftware/gui/pull/1110
* https://github.com/mendersoftware/deviceconnect/pull/13

ChangeLog:none
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Comment by lluiscampos** [comment]
As I said in the stand-up, this will require to bring `go-lib-micro` to Debian repositories. So we need to start working on that too.

**Comment by merlin-northern** [comment]
> As I said in the stand-up, this will require to bring `go-lib-micro` to Debian repositories. So we need to start working on that too.

yes, this is a good call! so good that you keep it under control.

File: session/session.go [hunk 1]
```diff
@@ -230,7 +230,7 @@ func MenderShellStopByUserId(userId string) (count uint, err error) {
 			continue
 		}
 		e := s.StopShell()
-		if e != nil {
+		if e != nil && procps.ProcessExists(s.shellPid) {
```
Comment (kacf) [inline session/session.go:233]: This looks like it should be in a different commit.

**Review by kacf — APPROVED** [review]
Very superficially, it looks ok to me, I'm not so familiar with the code though.

I spotted one thing below, but not sure if it's appropriate, I'll let you decide.

**Review by tranchitella — APPROVED** [review]

File: session/session.go (hunk 1 above)
Comment (merlin-northern) [reply session/session.go:233]: right! moved here.

**Comment by merlin-northern** [comment]
@kacf @tranchitella I added two small things:
* [one second sleep on reconnect](https://github.com/mendersoftware/mender-shell/pull/18/commits/9b06b9fafe9a7ecd907e68ceccec37278fb576ae) that one fixes 100% CPU usage on reconnect attempts
* [updated unit tests](https://github.com/mendersoftware/mender-shell/pull/18/commits/99f83440edf672b249dd37f1088b7bac6e422ec4) to increase coverage
please take a look again, the rest is the same, with Kristian suggestion applied.

File: app/daemon.go [hunk 2]
```diff
@@ -14,23 +14,25 @@
...
 	"github.com/mendersoftware/mender-shell/client/dbus"
 	"github.com/mendersoftware/mender-shell/client/mender"
 	configuration "github.com/mendersoftware/mender-shell/config"
+	"github.com/mendersoftware/mender-shell/connection"
 	"github.com/mendersoftware/mender-shell/deviceconnect"
 	"github.com/mendersoftware/mender-shell/procps"
 	"github.com/mendersoftware/mender-shell/session"
 	"github.com/mendersoftware/mender-shell/shell"
-	log "github.com/sirupsen/logrus"
-	"github.com/vmihailenco/msgpack"
```
Comment (tranchitella) [inline app/daemon.go]: Minor comment, would you mind rearranging the imports in two groups, non-mender-shell, and mender-shell specific?

**Review by tranchitella — APPROVED** [review]

File: app/daemon.go (hunk 2 above)
Comment (merlin-northern) [reply app/daemon.go]: sure thing!

Comment (merlin-northern) [reply app/daemon.go]: discussed over slack.

**Review by kacf — APPROVED** [review]
